public/app/features/admin/ServerStats.tsx

```diff
@@ -13,13 +13,15 @@ export const ServerStats = () => {
   const styles = useStyles2(getStyles);
 
   useEffect(() => {
-    getServerStats().then((stats) => {
-      setStats(stats);
-      setIsLoading(false);
-    });
+    if (contextSrv.hasAccess(AccessControlAction.ActionServerStatsRead, contextSrv.isGrafanaAdmin)) {
```

Comment:
Minor thing: I think it would be best to put that outside `useEffect`. So that we don't use the hook altogether if we don't have rights. Wdyt?

Reply:
Hooks can't be called conditionally, so the condition needs to be inside the hook :(